src/trait-bounds.md

```diff
@@ -156,6 +156,76 @@ fn call_on_ref_zero<F>(f: F) where F: for<'a> Fn(&'a i32) {
 }
 ```
 
+## Implied bounds
+
+Lifetime bounds required for types to be well-formed are sometimes inferred.
+
+```rust
+fn requires_t_outlives_a<'a, T>(x: &'a T) {}
+```
+The type parameter `T` is required to outlive `'a` for the type `&'a T` to be well-formed.
+This is inferred because the function signature contains the type `&'a T` which is
+only valid if `T: 'a` holds.
+
+Rust adds implied bounds for all parameters and outputs of functions. Inside of `requires_t_outlives_a`
```

Comment:
Still need to write this without referring to Rust directly. Everything is about Rust.